addons/mail/static/src/core/thread_service.js

```diff
@@ -665,9 +665,38 @@ export class ThreadService {
     }
 
     remove(thread) {
+        const { model, id } = thread;
+        const filteredStarredMessages = [];
+        let starredCounter = 0;
+        for (const msg of this.store.discuss.starred.messages) {
+            if (msg.resModel !== model || msg.resId !== id) {
+                starredCounter++;
+            } else {
+                filteredStarredMessages.push(msg);
+            }
```

Comment:
Isn't this condition inverted? The filtered messages are those that are different than `model` and `id`, and the counter must be decremented when they match.

Reply:
Oh crap. I missed it there. :sweat_smile: I will fix it with the failing test case. Thanks!